test_projects/cnode/test/example_test.exs

```diff
@@ -4,16 +4,86 @@ defmodule ExampleTest do
   test "init" do
     require Unifex.CNode
     assert {:ok, cnode} = Unifex.CNode.start_link(:example)
+
+    test_init(cnode)
+    test_unsigned_int(cnode)
+    test_string(cnode)
+    test_list(cnode)
+    test_list_as_string(cnode)
+    test_list_of_strings(cnode)
+    test_payload(cnode)
+    test_pid(cnode)
+    test_example_message(cnode)
+
+    test_undefined_function(cnode)
+    test_wrong_arguments(cnode)
+  end
+
+  def test_init(cnode) do
     assert :ok = Unifex.CNode.call(cnode, :init)
-    assert {:ok, 42, <<2, 2, 3>>} = Unifex.CNode.call(cnode, :foo, [self(), <<1, 2, 3>>])
-    assert_receive {:example_msg, 42}
+  end
+
+  def test_unsigned_int(cnode) do
+    assert {:ok, 0} = Unifex.CNode.call(cnode, :test_uint, [0])
+    assert {:ok, 5} = Unifex.CNode.call(cnode, :test_uint, [5])
+  end
+
+  def test_string(cnode) do
+    big_test_string = 'unifexunifexunifexunifexunifexunifexunifexunifexunifexunifexunifexunifexunifex
+    unifexunifexunifexunifexunifexunifexunifexunifexunifexunifexunifexunifexunifexunifexunifexunifex
+    unifexunifexunifexunifexunifexunifexunifexunifexunifexunifexunifexunifexunifexunifexunifexunifex
+    '
+    assert {:ok, ''} = Unifex.CNode.call(cnode, :test_string, [''])
+    assert {:ok, 'test_string'} = Unifex.CNode.call(cnode, :test_string, ['test_string'])
+    assert {:ok, '-12345'} = Unifex.CNode.call(cnode, :test_string, ['-12345'])
+    assert {:ok, '255'} = Unifex.CNode.call(cnode, :test_string, ['255'])
+    assert {:ok, big_test_string} = Unifex.CNode.call(cnode, :test_string, [big_test_string])
+  end
+
+  def test_list_as_string(cnode) do
+    0..253 |> Enum.each(fn x ->
+      list = [x, x+1, x+2]
+      assert {:ok, list} = Unifex.CNode.call(cnode, :test_list, [list])
+    end)
+    assert {:ok, [0, 0, 0]} = Unifex.CNode.call(cnode, :test_list, [[0, 0, 0]])
+    assert {:ok, [127, 127, 127]} = Unifex.CNode.call(cnode, :test_list, [[127, 127, 127]])
+    assert {:ok, [128, 128, 128]} = Unifex.CNode.call(cnode, :test_list, [[128, 128, 128]])
+    assert {:ok, [255, 255, 255]} = Unifex.CNode.call(cnode, :test_list, [[255, 255, 255]])
```

Comment:
Probably we don't need this cases because this numbers are tested above but something tells me they should be put here too.

Reply:
where exactly they're tested?

Reply:
Here 
https://github.com/membraneframework/unifex/blob/cef221788780157ef750571114c9031b7e62282a/test_projects/cnode/test/example_test.exs#L44-L47
as triples in form of [0, 1, 2] , [1, 2, 3] and so on. Each number between 0 and 255 will be tested.
Now I see that this code could be faster by testing [0,1,2], [3,4,5],... instead of [0,1,2], [1,2,3]...

Reply:
oh ok, well I believe that the lines 48-51 are enough, testing each possibility doesn't bring much more value while it has to be maintained